client/components/example/index.jsx

```diff
@@ -0,0 +1,171 @@
+
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import prism from 'prismjs';
+import 'prismjs/components/prism-jsx';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ButtonGroup from 'components/button-group';
+import Button from 'components/button';
+
+export default class Example extends Component {
+	static propTypes = {
+		renderButtons: PropTypes.func,
+	}
+
+	static defaultProps = {
+		renderButtons: noop,
+	}
+
+	constructor( props ) {
+		super( props );
+
+		this._toggleView = this._toggleView.bind( this );
+
+		this.state = {
+			sourceShown: false
+		};
+	}
+
+	_getButtonText() {
+		return this.state.sourceShown ? 'View Example' : 'View Source';
+	}
+
+	_renderButtons() {
+		const toggleViewButton = <Button
+			compact
+			primary
+			onClick={ this._toggleView }
+		>{ this._getButtonText() }</Button>;
+
+		return (
+			<div style={ { 'float': 'right' } }>
+				{ this.props.renderButtons() }
+				{ toggleViewButton }
+			</div>
+		);
+	}
+
+	_renderContent() {
+		const example = this.props.children;
+		const { sourceShown } = this.state;
+		const opacity = sourceShown ? 0 : 1;
+
+		return (
+			<div style={ { position: 'relative' } }>
+				<div style={ { opacity } }>{ example }</div>
+				{ this._renderSourceCode() }
+			</div>
+		);
+	}
+
+	_renderSourceCode() {
```

Comment:
What difference do you think it makes whether we check for `sourceShown` here or in its parent?